snapcraft/internal/pluginhandler/_runner.py

```diff
@@ -0,0 +1,193 @@
+# -*- Mode:Python; indent-tabs-mode:nil; tab-width:4 -*-
+#
+# Copyright (C) 2016, 2018 Canonical Ltd
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 3 as
+# published by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+import contextlib
+import json
+import os
+import shlex
+import subprocess
+import sys
+import tempfile
+import textwrap
+from typing import Any, Callable, Dict  # noqa
+
+from snapcraft.internal import (
+    common,
+    errors,
+)
+
+from ._scriptlet_functions import functions
+
+
+class Runner:
+    """The Runner class is responsible for orchestrating scriptlets."""
+
+    # FIXME: Need to quote builtin_functions typing because of
+    # https://github.com/python/typing/issues/259 which is fixed in Python
+    # 3.5.3.
+    def __init__(self, *, part_properties: Dict[str, Any], builddir: str,
+                 builtin_functions: 'Dict[str, Callable[..., None]]') -> None:
```

Comment:
try creating a type from `'Dict[str, Callable[..., None]]`, look at elf.py for inspiration

Reply:
I can't actually get it to work with the quoting, keeps telling me it's an invalid type.